File: CodeForces/101-200/141/A/141A.cpp
// Problem 141A: Amusing Joke
// https://codeforces.com/contest/141/problem/A
// Compiled with: g++ (Ubuntu 15.2.0-4ubuntu4) 15.2.0
// Linux version: Linux kernel 6.17.0-5-generic
// Submitted on: October 25th, 2025
// Codeforces language used: GNU G++23 14.2 (64 bit, msys2)

#include <iostream>
#include <string>
using namespace std;

int main() {
    string one;
    cin >> one;
    string two;
    cin >> two;
    string full;
    cin >> full;
    string correct = one + two;

    int letter_counts[26] = {};
    int value_of_a = 'A';

    for(char& c : correct) {
        letter_counts[c - value_of_a] += 1;
    }

    for(char& c : full) {
        letter_counts[c - value_of_a] -= 1;
    }

    for(int i = 0; i < 26; i++){
        if (letter_counts[i] != 0) {
            cout << "NO\n";
            return 0;
        }
    }
    cout << "YES\n";
}

File: CodeForces/101-200/141/A/141A.pl
# Problem 141A: Amusing Joke
# https://codeforces.com/contest/141/problem/A
# Executed with: perl 5, version 40, subversion 1 (v5.40.1) built for aarch64-linux-gnu-thread-multi
# Linux version: Linux kernel 6.17.0-6-generic
# Submitted on: October 27th, 2025
# Codeforces language used: Perl 5.20.1

#!/usr/bin/perl

use strict;
use warnings;

chomp(my $one = <STDIN>);
chomp(my $two = <STDIN>);
chomp(my $full = <STDIN>);

my $correct = $one . $two;

my %letter_counts;

# Count letters in the combined name
foreach my $c (split //, $correct) {
    $letter_counts{$c}++;
}

# Subtract letters from the pile
foreach my $c (split //, $full) {
    $letter_counts{$c}--;
}

# Check if all counts are zero
foreach my $val (values %letter_counts) {
    if ($val != 0) {
        print "NO\n";
        exit;
    }
}

print "YES\n";
